polyply/src/simple_seq_parsers.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2022 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import networkx as nx
+from vermouth.parser_utils import split_comments
+
+ONE_LETTER_AAS = {
+ "A": "ALA",
+ "R": "ARG",
+ "N": "ASN",
+ "D": "ASP",
+ "C": "CYS",
+ "Q": "GLB",
+ "E": "GLU",
+ "G": "GLY",
+ "H": "HIS",
+ "I": "ILE",
+ "L": "LEU",
+ "K": "LYS",
+ "M": "MET",
+ "F": "PHE",
+ "P": "PRO",
+ "O": "PLY",
+ "S": "SER",
+ "U": "SEC",
+ "T": "THR",
+ "W": "TRP",
+ "Y": "TYR",
+ "V": "VAL",
+ "B": "ASX",
+ "Z": "GLX",
+ "X": "XAA",
+ "J": "XLE",}
+
+ONE_LETTER_DNA = {"A": "DA",
+                  "C": "DC",
+                  "G": "DG",
+                  "T": "DT"}
+
+ONE_LETTER_RNA = {"A": "A",
+                  "C": "C",
+                  "G": "G",
+                  "T": "U"}
+
+def _monomers_to_linear_nx_graph(monomers):
+    seq_graph = nx.Graph()
+    seq_graph.add_edges_from(zip(range(0, len(monomers))))
+    nx.add_node_attributes(seq_graph, {node: resname for node, resname in zip(seq_graph.nodes, monomers)}, "resname")
+    nx.add_node_attributes(seq_graph, {node: node+1 for node in zip(seq_graph.nodes)}, "resid")
+    return seq_graph
+
+def parse_plain_delimited(lines, delimiter=" "):
+    """
+    Parse a plain delimited text file. The delimiter can
+    be any special character.
+    """
+    monomers = []
+    for line in lines:
+        for resname in line.strip().split(delimiter):
+            monomers.append(resname)
+    seq_graph =  _monomers_to_linear_nx_graph(monomers)
+    return seq_graph
+
+def parse_plain(lines, DNA=False, RNA=False):
```

Comment:
@pckroon: I'm implementing some parsers for simple sequence files. Would you pass the lines of the file to the parser or the handle and then open the file in the parser itself?

Reply:
I usually pass an iterator of lines (opened file or list of lines, the function does not need to know). If I need to pass a file name for convenience, I have another function that is just a very thin wrapper around the first function.

Reply:
pretty much like it is now?

Reply:
yes